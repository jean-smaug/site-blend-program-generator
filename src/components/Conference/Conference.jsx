import React from 'react';
import PropType from 'prop-types';

const Conference = ({ name }) => (
  <div>
    <button className="switchButton">Switch</button>
    <h4>{name}</h4>
    <button className="lockButton">Lock</button>
  </div>
);

Conference.propTypes = {
  name: PropType.string.isRequired,
};

export default Conference;
