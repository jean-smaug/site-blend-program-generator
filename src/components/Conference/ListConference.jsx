import React from 'react';
import PropType from 'prop-types';

import Conference from './Conference';

const ListConference = ({ day }) => {
  const { eight, ten, fourteen, sixteen } = day;

  return (
    <div>
      <Conference {...eight[0]} />
      <Conference {...ten[0]} />
      <Conference {...fourteen[0]} />
      <Conference {...sixteen[0]} />
    </div>
  );
};
ListConference.propTypes = {
  day: PropType.shape({
    eight: Conference.propTypes,
    ten: Conference.propTypes,
    fourteen: Conference.propTypes,
    sixteen: Conference.propTypes,
  }).isRequired,
};

export default ListConference;
