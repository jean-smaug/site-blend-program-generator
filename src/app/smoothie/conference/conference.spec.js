import React from 'react';
import { shallow } from 'enzyme';
import Conference from './conference.component';

const props = {
  name: 'foo',
  timeBegin: '10',
  timeEnd: '12',
};

describe('conference.component', () => {
  it('should render conference.component', () => {
    const conference = shallow(<Conference {...props} />);
    expect(conference.getNodes()).toMatchSnapshot();
  });
});
