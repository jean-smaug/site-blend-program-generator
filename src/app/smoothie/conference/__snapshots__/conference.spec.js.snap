// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conference.component should render conference.component 1`] = `
Array [
  <div
    onClick={[Function]}
    role="presentation"
>
    <div
        className="columns"
    >
        <div
            className="column"
        >
            <div
                className="conference"
            >
                <div
                    className="conference-opt"
                >
                    <span
                        className="conference-time"
                    >
                        10h00 &gt; 12h00
                    </span>
                </div>
                <div
                    className="conference-title"
                    role="button"
                >
                    Temps libre
                </div>
            </div>
        </div>
    </div>
</div>,
]
`;
